datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -18,17 +18,74 @@ package kafka
 
 import (
 	"context"
+	"encoding/binary"
 	"errors"
 	"fmt"
 
 	"github.com/Shopify/sarama"
 )
 
-var eventNameErrorMsg = "the event name must be either 'blockgroup' or 'tracegroup'"
+//go:generate mockgen -destination=./mocks/consumer_group_session_mock.go -package=mocks github.com/klaytn/klaytn/datasync/chaindatafetcher/kafka ConsumerGroupSession
+// ConsumerGroupSession is for mocking sarama.ConsumerGroupSession for better testing.
+type ConsumerGroupSession interface {
+	MarkOffset(topic string, partition int32, offset int64, metadata string)
+	MarkMessage(msg *sarama.ConsumerMessage, metadata string)
+}
+
+var (
+	eventNameErrorMsg          = "the event name must be either 'blockgroup' or 'tracegroup'"
+	nilConsumerMessageErrorMsg = "the given message should not be nil"
+	wrongHeaderNumberErrorMsg  = "the number of header is not expected"
+	wrongHeaderKeyErrorMsg     = "the header key is not expected"
+	wrongSegmentErrorMsg       = "there is a missing segment"
+	noHandlerErrorMsg          = "the handler does not exist for the given topic"
+	emptySegmentErrorMsg       = "there is no segment in the segment slice"
+)
 
 // TopicHandler is a handler function in order to consume published messages.
 type TopicHandler func(message *sarama.ConsumerMessage) error
 
+// Segment represents a message segment with the parsed headers.
+type Segment struct {
+	orig  *sarama.ConsumerMessage
+	key   string
+	total uint64
+	index uint64
```

Comment:
What do `total` and `index` mean?

Reply:
`total` is the total number of segments of the original message, and `index` is the position of the segment in the original message. Is it unclear?

Reply:
It would be nice if your above comments are written in the code 😄

Reply:
The variable names will be refactored in another PR.